nullaway/src/main/java/com/uber/nullaway/Nullness.java

```diff
@@ -128,73 +128,85 @@ public String toString() {
     return displayName;
   }
 
-  private static boolean hasNullableAnnotation(Stream<? extends AnnotationMirror> annotations) {
+  private static boolean hasNullableAnnotation(
+      Stream<? extends AnnotationMirror> annotations, Config config) {
     return annotations
         .map(anno -> anno.getAnnotationType().toString())
-        .anyMatch(Nullness::isNullableAnnotation);
+        .anyMatch(anno -> isNullableAnnotation(anno, config));
   }
 
-  private static boolean hasNonNullAnnotation(Stream<? extends AnnotationMirror> annotations) {
+  private static boolean hasNonNullAnnotation(
+      Stream<? extends AnnotationMirror> annotations, Config config) {
     return annotations
         .map(anno -> anno.getAnnotationType().toString())
-        .anyMatch(Nullness::isNonNullAnnotation);
+        .anyMatch(anno -> isNonNullAnnotation(anno, config));
   }
 
   /**
    * @param annotName annotation name
    * @return true if we treat annotName as a <code>@Nullable</code> annotation, false otherwise
    */
-  public static boolean isNullableAnnotation(String annotName) {
+  public static boolean isNullableAnnotation(String annotName, Config config) {
     return annotName.endsWith(".Nullable")
         // endsWith and not equals and no `org.`, because gradle's shadow plug in rewrites strings
         // and will replace `org.checkerframework` with `shadow.checkerframework`. Yes, really...
         // I assume it's something to handle reflection.
-        || annotName.endsWith(".checkerframework.checker.nullness.compatqual.NullableDecl");
+        || annotName.endsWith(".checkerframework.checker.nullness.compatqual.NullableDecl")
+        || (config.acknowledgeAndroidRecent()
+            && annotName.equals("androidx.annotation.RecentlyNullable"));
```

Comment:
We could also match on `.RecentlyNullable` like we do for `.Nullable` and catch, e.g. shadowed code and other libraries following the same convention. But let's defer that until we see an actual issue with the full match :)

That said I think there are `RecentlyNullable` annotations in the Android support library with namespaces from before the switch to the `androidx` namespace. See: [`android.support.annotation.RecentlyNonNull`](https://android.googlesource.com/platform/tools/metalava/+/fcb3d99ad3fe17a266d066b28ceb6c526f7aa415/stub-annotations/src/main/java/android/support/annotation/RecentlyNonNull.java) Not sure if there are others... (e.g. `v4` or `v7` namespaces)

Reply:
Those only exist because the androidx versions were automatically rewritten. The Android APIs will only reference the androidx versions and the Kotlin compiler only honors them as well: https://github.com/JetBrains/kotlin/blob/12736edfc1716e0d617ca47411885ddd2333ba1c/core/descriptors.jvm/src/org/jetbrains/kotlin/load/java/JvmAnnotationNames.kt#L56-L57

Reply:
Ok, I was unaware.  Do you have a suggestion on what we should do here then?  Should we just match on `.RecentlyNullable`?  I'd rather not add a regex match in here.

Reply:
nvm, based on @JakeWharton's comment I guess we can just look for the androidx version?